internal/mongo/event.go

```diff
@@ -0,0 +1,50 @@
+package mongo
+
+import (
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+const documentKeyField = "documentKey"
+const idField = "_id"
+
+// check if a change event if valid
+// A valid event has at least a document key value
+func isValid(event bson.M) error {
+	_, err := documentID(event)
+	return err
+}
+
+// marshall event to an array of bytes
+func marshall(event bson.M) ([]byte, error) {
+	if err := isValid(event); err != nil {
+		return nil, fmt.Errorf("%v is not a valid change event : %w", event, err)
+	}
+	return bson.MarshalExtJSON(event, true, true)
```

Comment:
I think it's better to user bson marshaller vs json marshaller.
What do you think ?

```
{
  "_id": {
    "_id": {
      "$oid": "5e63a5282a425a6e386e6a0c"
    },
    "copyingData": true
  },
  "operationType": "insert",
  "ns": {
    "db": "intref001",
    "coll": "video"
  },
  "documentKey": {
    "_id": {
      "$oid": "5e63a5282a425a6e386e6a0c"
    }
  },
  "fullDocument": {
    "ref_id": "818c8a01-567f-4a35-a1b7-d8a6b9350ebe",
    "title": "Title 4459961176045108197",
    "duration": {
      "$numberLong": "5973000"
    },
    "type": {
      "$numberInt": "0"
    },
    "_id": {
      "$oid": "5e63a5282a425a6e386e6a0c"
    }
  }
}
```

vs

```
{
  "_id": {
    "_id": "5e63a5282a425a6e386e6a0c",
    "copyingData": true
  },
  "documentKey": {
    "_id": "5e63a5282a425a6e386e6a0c"
  },
  "fullDocument": {
    "_id": "5e63a5282a425a6e386e6a0c",
    "duration": 5973000,
    "ref_id": "818c8a01-567f-4a35-a1b7-d8a6b9350ebe",
    "title": "Title 4459961176045108197",
    "type": 0
  },
  "ns": {
    "coll": "video",
    "db": "intref001"
  },
  "operationType": "insert"
}
```

Reply:
Maybe we should benchmark both codes ;) not sure the payoff is that important, but despite on readability/maintainability it should have a better performance